### PR TITLE
fix: remove ts expect error

### LINKED ISSUE #1811: [Bug]: Reviewing usage of  `@ts-expect-error`, and replace with `@ts-ignore`  if we misused before.
### Reproduction link or steps

Search `@ts-expect-error` in our codebase.

### What is expected?

Replace it, if it is misused before.

### What is actually happening?

above

### System Info

```Shell
n/a
```

### Any additional comments?

ref: 
https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-9.html#ts-ignore-or-ts-expect-error

### PR DESCRIPTION
**Pull request by IWANABETHATGUY**
### Description
Closed #1811

### CODE REVIEW

File: packages/rolldown/src/utils/normalize-hook.ts [hunk 1]
```diff
@@ -1,25 +1,20 @@
...
+export function normalizeHook<T extends ObjectHook<Function>>(
+	hook: T,
+): T extends ObjectHook<infer A, infer B>
+	? T extends Function
+		? [A, {}]
+		: [A, Omit<B, "handler">]
+	: never {
+	if (typeof hook === "function") {
+    // @ts-ignore
+		return [hook, {}];
```
Comment (IWANABETHATGUY) [inline packages/rolldown/src/utils/normalize-hook.ts]: previously: 
image
current: 
image

Comment (IWANABETHATGUY) [reply packages/rolldown/src/utils/normalize-hook.ts]: Sadly, it still needs ts **directive**, forgive me poor **type gymnastics**

File: packages/rolldown/src/utils/normalize-hook.ts [hunk 2]
```diff
@@ -1,25 +1,19 @@
...
-
-export function normalizeHook<H extends ObjectHook<AnyFn, AnyObj>>(
-  hook: H,
-): H extends ObjectHook<infer Handler, infer Options>
-  ? [Handler, NotFn<Options>]
+export function normalizeHook<T extends ObjectHook<Function>>(
+  hook: T,
+): T extends ObjectHook<infer A, infer B>
+  ? T extends Function
+    ? [A, {}]
```
Comment (underfin) [inline packages/rolldown/src/utils/normalize-hook.ts:7]: I refactor the function to make support `ObjectHook<string, {}>` for AddonHooks, and remove using `infer` make typing clear, the `@ts-expect-error` is not used.

Comment (IWANABETHATGUY) [reply packages/rolldown/src/utils/normalize-hook.ts:7]: I see the pr, but it lost the ability to infer the type

Comment (IWANABETHATGUY) [reply packages/rolldown/src/utils/normalize-hook.ts:7]: And it may cause more type errors in the down stream 
image

**Comment by IWANABETHATGUY** [comment]
I am gonna merge this, but I am open to accept if any one have a better idea

**Pull request merged**